solr/core/src/java/org/apache/solr/response/TextQueryResponseWriter.java

```diff
@@ -81,7 +83,8 @@ private static Writer buildWriter(OutputStream outputStream, String charset)
    *
    * <p>See SOLR-8669.
    */
-  private static class NonFlushingStream extends OutputStream {
+  // nocommit discuss moving to SolrDispatchFilter wrapper.  If keep them move?
```

Comment:
This is an observation I had to do things better.  Basically, don't we want Solr to always prevent a flush, not just specifically here only sometimes for some QueryResponseWriters?  If so, it can easily go on the close shield thing (see elsewhere in this PR to show how.  Then we can drop this NonFlushingStream thing and not worry about wether we should use it in other scenarios.  CC @magibney wonder if you have thoughts on this one